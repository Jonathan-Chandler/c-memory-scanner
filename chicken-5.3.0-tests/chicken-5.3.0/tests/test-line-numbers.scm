(define-syntax bar
  (er-macro-transformer
   (lambda (e inj cmp) (get-line-number (cadr e)))))

(define-syntax foo
  (ir-macro-transformer
   (lambda (e inj cmp) (get-line-number (cadr e)))))

(assert (equal? "test-line-numbers.scm:9" (the * (foo (hello-ir)))))
(assert (equal? "test-line-numbers.scm:10" (the * (bar (hello-er)))))

(define-syntax baz
  (ir-macro-transformer
   (lambda (e inj cmp)
     (cadr e))))

(define (f)
  (baz
   (+ 'a)))
