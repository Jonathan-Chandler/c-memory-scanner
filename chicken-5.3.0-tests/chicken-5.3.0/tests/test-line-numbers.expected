
Warning: Invalid argument
  In file `test-line-numbers.scm:19',
  In procedure `f',
  In procedure call:

    (scheme#+ 'a)

  Argument #1 to procedure `+' has an invalid type:

    symbol

  The expected type is:

    number

  This is the expression:

    'a

  Procedure `+' from module `scheme' has this type:

    (#!rest number -> number)
